roofit/roostats/src/MCMCInterval.cxx

```diff
@@ -449,8 +452,12 @@ void MCMCInterval::CreateDataHist()
       return;
    }
 
-   fDataHist = std::unique_ptr<RooDataHist>{fChain->GetAsDataHist(SelectVars(fParameters),
-         EventRange(fNumBurnInSteps, fChain->Size()))}.release();
+   const RooDataSet* fChainData = fChain->GetAsConstDataSet();
+   std::unique_ptr<RooAbsData> data{fChainData->reduce(SelectVars(fParameters), EventRange(fNumBurnInSteps, fChain->Size()))};
+
+   fDataHist = std::unique_ptr<RooDataHist>{
+      RooFit::makeOwningPtr(std::unique_ptr<RooDataHist>{
+         static_cast<RooDataSet &>(*data).binnedClone()})}.release();
```

Comment:
```suggestion
   fDataHist = static_cast<RooDataSet &>(*data).binnedClone();
```
Doesn't need to be so complicated here, since `fDataHist` needs to be a raw pointer anyway.